sbpy/data/core.py

```diff
@@ -934,3 +938,135 @@ def verify_fields(self, field=None):
+        `~astropy.time.Time` object, or a number, `~astropy.units.Quantity`
+        object, or string that can be inferred to be a time by the existing
+        column of the same name or by its position in the sequence.  In
+        this case, the type of time values must be valid to initialize
+        an `~astropy.time.Time` object with format='jd' or 'isot', and
+        the scale of time is default to the scale of the corresponding
+        existing column of time.
+
+        Examples
+        --------
+        >>> import astropy.units as u
+        >>> from sbpy.data import DataClass
+        >>>
+        >>> data = DataClass.from_dict(
+        ...         {'rh': [1, 2, 3] * u.au, 'delta': [1, 2, 3] * u.au})
+        >>> row = {'rh': 4 * u.au, 'delta': 4 * u.au, 'phase': 15 * u.deg}
+        >>> data.add_row(row)
+        """
+        if isinstance(vals, Row):
+            vals = DataClass.from_table(vals)
+        else:
+            if isinstance(vals, Mapping):
+                keys_list = list(vals.keys())
+                vals_list = [vals[k] for k in keys_list]
+                vals = vals_list
+                if names is None:
+                    names = keys_list
+            else:
+                # assume it's an iterable that can be taken as columns
+                if names is None:
+                    # if names of columns are not specified, default to the
+                    # existing names and orders
+                    names = self.field_names
+            # check if any astropy Time columns
+            for i, k in enumerate(names):
+                if k in self and isinstance(self[k], Time):
+                    vals[i] = Time(vals[i], scale=self[k].scale,
+                                   format='isot' if isinstance(vals[i], str)
+                                   else 'jd')
+            vals = DataClass.from_rows(vals, names, units=units)
+        self.join(vals)
+
+    def add_column(self, col, name=None, unit=None):
+        """Add a new column
+
+        col : `~astropy.table.Column` object, or sequence
+            The column to be added.  Must have the same length as the
+            existing table.
+        name : array-like str, optional
+            Specify the name of added column.  If not ``None``, then
+            takes precedence over ``col.name``.
+        unit : array-like `~astropy.units` objects or str, optional
+            Unit to be applied to the new column.  If not ``None``, then
+            take precedence over ``col.unit``.
+        """
+        if name is None:
+            name = getattr(col, 'name', '')
+        if unit is None:
+            unit = getattr(col, 'unit', None)
+        if name in self:
+            raise DataClassError('Column {} already exists.'.format(name))
+        self.apply(col, name=name, unit=unit)
+
+    def join(self, data):
+        """Join another DataClass object to the end of DataClass
```

Comment:
I think this is vertical stacking, i.e., `vstack`.  As I understand it, table joining is a different operation, borrowed from SQL, and according to wikipedia, borrowed from relational algebra: https://en.wikipedia.org/wiki/Join_(SQL)  It is a very powerful way to combine tables, so I'd rather rename this as `vstack` and reserve `join` for a future implementation of `astropy.table.join`.